src/stack/stack.jl

```diff
@@ -307,24 +307,45 @@ end
     DimStack <: AbstractDimStack
 
     DimStack(data::AbstractDimArray...; kw...)
-    DimStack(data::Tuple{Vararg{AbstractDimArray}}; kw...)
-    DimStack(data::NamedTuple{Keys,Vararg{AbstractDimArray}}; kw...)
-    DimStack(data::NamedTuple, dims::DimTuple; metadata=NoMetadata(); kw...)
+    DimStack(data::Union{AbstractArray,Tuple,NamedTuple}[, dims::DimTuple]; kw...)
 
 DimStack holds multiple objects sharing some dimensions, in a `NamedTuple`.
 
-Notably, their behaviour lies somewhere between a `DimArray` and a `NamedTuple`:
+## Arguments
 
-- indexing with a `Symbol` as in `dimstack[:symbol]` returns a `DimArray` layer.
-- iteration and `map` apply over array layers, as indexed with a `Symbol`.
+- `data`: `AbstractDimArray` or an `AbstractArray`, `Tuple` or `NamedTuple` of `AbstractDimArray`s or `AbstractArray`s.
+- `dims`: `DimTuple` of `Dimension`s. Required when `data` is not `AbstractDimArray`s.
+
+## Keywords
+
+- `name`: `Array` `Tuple` of `Symbol` names for each layer. By default
+    the names of `DimArrays` are used, or `:layer1`, `:layer2`, etc.
```

Comment:
```suggestion
- `name`: `Array` or `Tuple` of `Symbol` names for each layer. By default
    the names of `DimArrays` are or keys of a `NamedTuple` are used, 
    or `:layer1`, `:layer2`, etc.
```
We could also consider just requiring this to be Tuple of Symbol imo.

Reply:
Especially as passing a Vector of names isn't untested at the moment.

Reply:
A vector is useful when there is 300 of them